pkg/cri/server/container_create_linux.go

```diff
@@ -307,7 +307,7 @@ func (c *criService) containerSpecOpts(config *runtime.ContainerConfig, imageCon
 	specOpts = append(specOpts, customopts.WithAdditionalGIDs(userstr))
 
 	apparmorSpecOpts, err := generateApparmorSpecOpts(
-		securityContext.GetApparmorProfile(),
+		securityContext.GetApparmorProfile(), //lint:ignore SA1019 ignore deprecated warning
```

Comment:
Looks like this was deprecated in https://github.com/kubernetes/kubernetes/commit/7b9d7fd953f43428fac0b030f3ff2a269fc9afcc (part of https://github.com/kubernetes/kubernetes/pull/95876)

Can we replace this for `GetApparmor`, and use the new type? (or at least add a TODO and a tracking issue)?

Reply:
yep, will switch to the new API, hence the WIP. Also it's weird that the `lint:ignore` is being ignored!

Reply:
Can you try;

```suggestion
		securityContext.GetApparmorProfile(), //nolint:staticcheck // ignore SA1019 ignore deprecated warning
```

(looked up how we ignore these in the docker code, and that seems to work there)

Reply:
Looks like @mikebrow wrapped up these in https://github.com/containerd/containerd/pull/4700 yay!